optuna/study.py

```diff
@@ -290,6 +291,15 @@ def optimize(
                     _iter = iter(int, 1)
 
                 with Parallel(n_jobs=n_jobs, prefer="threads") as parallel:
+                    if not isinstance(parallel._backend, joblib.parallel.ThreadingBackend) and \
```

Comment:
A detail but I believe this is the public interface for querying the backend, [`joblib.parallel.get_active_backend`]( https://github.com/joblib/joblib/blob/master/joblib/parallel.py#L81).

Reply:
Thank you for your comment.
Actually, I also checked the function before, but I found out that it returned the [`default backend`](https://github.com/joblib/joblib/blob/master/joblib/parallel.py#L82) instead of the active backend despite the name.

A simple reproduction code is as follows:

```python
>>> import joblib
>>> with joblib.parallel.Parallel(prefer='threads', n_jobs=2) as parallel:
...     print(parallel._backend)
...     print(joblib.parallel.get_active_backend())
...
<joblib._parallel_backends.ThreadingBackend object at 0x1208f74e0>
(<joblib._parallel_backends.LokyBackend object at 0x1208f75c0>, 1)
```

The actual backend is threading but `joblib.parallel.get_active_backend()` returns loky.
It means that users see the warning message even if they use threading.

I'm not sure if this is expected behavior of `joblib.parallel.get_active_backend()`, but I think we cannot use it for now.

I fixed the test case to take care of the difference between `parallel._backend` and `joblib.parallel.get_active_backend()` in this commit (https://github.com/optuna/optuna/pull/780/commits/7acda9e6b55a53ea3f7b4530f1d1ca60b3047016).

Reply:
Oh I see, I should've read the documentation properly. Thanks for the detailed explanation.